### DISCUSSION TITLE
[Bug]: Unregister from useForm doesn't work anymore

### DISCUSSION THREAD

**Discussion by flotos** [react-hook-form/react-hook-form#6401] [Q&A] [answered]
### Version Number

7.14.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-unregister-v6-forked-3d08f?file=/src/index.js

### Steps to reproduce

1. Go to the codesandbox provided
2. Fill the form
3. Click on "unregister lastname"
4. click on submit
5. the alert dialog still shows lastname's content

### Expected behaviour

The alert dialog at the end should show an object with no "lastname" property according to the docs : https://react-hook-form.com/api/useform/unregister

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
```tsx
<input {...register("lastName")} />
<button
  type="button"
  onClick={() => {
    unregister("lastName");
  }}
>
  unregister lastName
</button>
```

you should unregister that input and also remove that input as well, or else it will get `register` again.

**Reply by flotos** [reply]
Alright thanks.
The codesandbox I provided is fork of the one in the doc, with the only difference I have updated the package version. (The original version had the same problem on 7.2.2). I think it should then be updated with this answer too.

**Reply by flotos** [reply]
Here is an example working one : https://codesandbox.io/s/react-hook-form-unregister-v6-forked-3d08f?file=/src/index.js
